_posts/2017-10-04-scalamacros.md

```diff
@@ -0,0 +1,299 @@
+retire the scala.reflect-based macro system.
+Another justification for retiring the scala.reflect-based macro system was
+that a new macro system based on Scalameta was "just around the corner".
+
+### v2: scala.meta
+
+The [Scalameta] project was founded to become a better macro system for Scala,
+with the vision to replace scala.reflect as the de-facto metaprogramming
+toolkit for Scala.
+
+With Scalameta, we managed to support macro annotations in
+[Scala 2.x](http://scalameta.org/paradise/),
+[IntelliJ Scala plugin](https://blog.jetbrains.com/scala/2016/11/11/intellij-idea-2016-3-rc-scala-js-scala-meta-and-more/)
+and
+[Dotty](https://github.com/liufengyun/eden).
+The novelty with Scalameta macros was that they converted compiler-specific
+ASTs into the [Scalameta AST], which is a large collection "dumb" data
+containers that leak no implementation details from the compiler.
+The details of this approach are explained in more detail in [SIP-29], a
+proposal to use Scalameta as the foundation for building macros in Scala.
+
+In Scalafmt, I use Scalameta macro annotations to generate readers for
+over 
+[90 different configuration options](http://scalameta.org/scalafmt/#Configuration)
+into a single case class
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">New style &quot;inline/meta&quot; macro merged into scalafmt, Dotty-ready macros are happening and they are amazing! <a href="https://twitter.com/hashtag/scala?src=hash">#scala</a> <a href="https://github.com/scalameta/scalafmt/pull/459">scalameta/scalafmt#459</a> <a href="https://t.co/ZvjdWE0Gp3">pic.twitter.com/ZvjdWE0Gp3</a></p>&mdash; Ólafur Páll Geirsson (@olafurpg) <a href="https://twitter.com/olafurpg/status/779372897198637057">September 23, 2016</a></blockquote>
+
+Many other OSS Scala libraries also use Scalameta macro annotations to
+to maximize type safety, expressiveness and performance:
+- [Freestyle](http://frees.io/):  cohesive & pragmatic framework of FP centric
+  Scala libraries
+- [Stalagmite](https://vovapolu.github.io/scala/stalagmite/perf/2017/09/02/stalagmite-performance.html):
+  effective and customizable replacement of conventional case classes with
+  convenient optimizations.
+- [Mainecoon](http://kailuowang.com/mainecoon/): A library for transforming and
+  composing tagless final algebras
+- [Dilate](https://github.com/vitorsvieira/dilate) /
+  [Newtypes](https://github.com/alexknvl/newtypes): better value
+  classes/newtypes for Scala.
+- [Example.scala](https://static.javadoc.io/com.thoughtworks.example/unidoc_2.12/2.0.0/com/thoughtworks/example.html):
+  Generate unit tests from Scaladoc strings
+
+The Scala community is indeed creative and eager to explore new metaprogramming
+facilities.
+
+Scalameta macro annotations are far from perfect, they suffer from integration
+problems with Scaladoc, Scala IDE/presentation compiler, 
+Scala REPL and other compiler plugins such as Scoverage.
```

Comment:
...E.g?

Reply:
Not sure I follow, but I updated the text to use e.g. :)